### PR TITLE
hw(5): Add various tests

### PR DESCRIPTION
**Pull request by dbernhard-0x7CD**

### CODE REVIEW

File: 05/dbernhard/err_function_return_type.oat [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  return false;
+}
+
+C get() {
+  return new C { c = 2 };
+}
+
+int program(int argc, string[] argv) {
+  if (same_x(get(), get())) {
```
Comment (nicdard) [inline 05/dbernhard/err_function_return_type.oat:19]: IMHO, I would say the test is more about wrong passed parameters type (C is not a subtype of A) than wrong function return type, but it is a point of view so doesn't really matter

Comment (nicdard) [inline 05/dbernhard/err_function_return_type.oat:19]: IMHO, I would say the test is more about wrong passed parameters type (C is not a subtype of A) than wrong function return type, but it is a point of view so doesn't really matter

**Review by nicdard — CHANGES_REQUESTED** [review]

File: 05/dbernhard/err_function_return_type.oat (hunk 1 above)
Comment (dbernhard-0x7CD) [reply 05/dbernhard/err_function_return_type.oat:19]: I'd leave it as it is to check if the subtype is correctly implemented.

Comment (nicdard) [reply 05/dbernhard/err_function_return_type.oat:19]: Sure makes sense

**Comment by nicdard** [comment]
Thanks!

**Pull request merged**